code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -145,13 +145,15 @@
 
 /mob/living/simple_animal/Life()
 	..()
-	life_tick++
-	//Health
-	updatehealth()
 
+	life_tick++
 	if(stat == DEAD)
 		return 0
 
+
+	//Health
+	updatehealth()

```

Comment:
updatehealth was in the wrong place, causing it to be called endlessly on dead animals